Sources/ApodiniDatabase/Context/DatabaseInjectionContext.swift

```diff
@@ -0,0 +1,18 @@
+import Foundation
+import Fluent
+
+///A Protocol which provides info about the expected the type of a `FieldKey`.
+protocol DatabaseInjectionContext {
+    ///A `Fluent.FieldKey`
+    var key: FieldKey { get }
+    ///The expected type for the fieldkey
+    var value: TypeContainer { get }
+}
+
+///A struct implementing `DatabaseInjectionContext` and containing a fieldkey and the expected type for that key.
+struct ModelInfo: DatabaseInjectionContext, Equatable {
```

Comment:
I would suggest to remove the conformance to `Equatable` as there is no implementation in the type anymore and it doesn't seem to be sued anywhere.